hazelcast/src/test/java/com/hazelcast/test/starter/HazelcastStarter.java

```diff
@@ -243,4 +224,24 @@ private static File getOrCreateVersionDirectory(String versionSpec) {
             return urls;
         }
     }
+
+    /**
+     * Creates a temporary directory for downloaded artifacts.
+     *
+     * @param versionSpec the version specification, which should include both version and enterprise edition indication,
+     *                    e.g. "3.8-EE", "3.8.1"
+     */
+    public static File getOrCreateVersionDirectory(String versionSpec) {
+        File workingDir = WORKING_DIRECTORY;
+        if (!workingDir.isDirectory() || !workingDir.exists()) {
+            throw new GuardianException("Working directory " + workingDir.getAbsolutePath() + " does not exist.");
+        }
+
+        File versionDir = new File(WORKING_DIRECTORY, versionSpec);
+        // the versionDir may already exist when the same Hazelcast version artifacts were previously downloaded
+        if (!versionDir.exists() && !versionDir.mkdir()) {
```

Comment:
This is the actual fix, only attempt `mkdir` if directory doesn't exist already. Method was made `public` for test.